app/services/sso/jwt.rb

```diff
@@ -24,6 +27,18 @@ def authenticated?
       self.user = User.current.presence || authenticate!
     end
 
+    def allowed_to?(action)
+      return false if @current_user.disabled?
+      return true if @current_user.admin?
+
+      required_permissions = Foreman::AccessControl.permissions_for_controller_action(action).map(&:name)
```

Comment:
Does this mean that if a user does not have a specific permission but the JWT token lists the permission, the user is granted the action regardless that he does not has the permission in the first place.